tracing/zipkin/http_test.go

```diff
@@ -0,0 +1,160 @@
+package zipkin_test
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"reflect"
+	"testing"
+
+	zipkinkit "github.com/go-kit/kit/tracing/zipkin"
+	kithttp "github.com/go-kit/kit/transport/http"
+	zipkin "github.com/openzipkin/zipkin-go"
+	"github.com/openzipkin/zipkin-go/model"
+	"github.com/openzipkin/zipkin-go/reporter/recorder"
+)
+
+const (
+	testName     = "test"
+	testBody     = "test_body"
+	testTagKey   = "test_key"
+	testTagValue = "test_value"
+)
+
+func TestHttpClientTracePropagatesParentSpan(t *testing.T) {
+	rec := recorder.NewReporter()
+	defer rec.Close()
+
+	tr, _ := zipkin.NewTracer(rec)
+
+	rURL, _ := url.Parse("http://test.com")
+
+	c := kithttp.NewClient(
+		"GET",
+		rURL,
+		func(ctx context.Context, r *http.Request, i interface{}) error {
+			return nil
+		},
+		func(ctx context.Context, r *http.Response) (response interface{}, err error) {
+			return nil, nil
+		},
+	)
+
+	clientOption := zipkinkit.HTTPClientTrace(tr)
+	clientOption(c)
+
+	parentSpan := tr.StartSpan("test")
+
+	ctx := zipkin.NewContext(context.Background(), parentSpan)
+
+	_, err := c.Endpoint()(ctx, nil)
```

Comment:
here you see why it is more idiomatic to create an endpoint instead of a client. an endpoint can be wrapped with middleware like circuitbreakers, retry logic, etc.
```
sum := kithttp.NewClient(...).Endpoint()

// wrap middlewares
sum = someMiddleware(middlewareSettings)(sum)
```